homeassistant/components/supla/light.py

```diff
@@ -0,0 +1,175 @@
+            brightness = min(100, ceil(kwargs[ATTR_BRIGHTNESS] * 100 / 255.0))
+            await self.async_action("SET_RGBW_PARAMETERS", brightness=brightness)
+            return
+        await self.async_action("TURN_ON")
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        await self.async_action("TURN_OFF")
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if switch is on."""
+        if state := self.channel_data.get("state"):
+            return state["on"]
+        return False
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        state = self.channel_data.get("state")
+        if not state or "brightness" not in state:
+            return None
+        brightness = state.get("brightness")
+        # Convert from (0-100) to Home Assistant brightness (0-255)
+        return min(255, ceil(brightness * 255 / 100.0))
+
+
+class SuplaRGBLightingEntity(SuplaEntity, LightEntity):
+    """Representation of a Supla RGB light."""
+
+    _attr_color_mode = ColorMode.HS
+    _attr_supported_color_modes = {ColorMode.HS}
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS) or self.brightness
+        hs_color = kwargs.get(ATTR_HS_COLOR) or self.hs_color
+        if not self.is_on and ATTR_HS_COLOR in kwargs and ATTR_BRIGHTNESS not in kwargs:
+            # When the light is off, brightness state is set to 0.
+            # However, last color state is preserved, and we can extract last brightness from it.
+            brightness = self._brightness_from_color
+
+        if not kwargs or not (hs_color and brightness):
+            await self.async_action("TURN_ON")
+            return
+        rgb_update_color = color_util.color_hsv_to_RGB(
+            hs_color[0], hs_color[1], brightness / 255.0 * 100
+        )
+        (red, green, blue) = rgb_update_color
+        # Color is set as the hex string of the form "0xRRGGBB"
+        color = "0x" + color_util.color_rgb_to_hex(red, green, blue)
+        await self.async_action("SET_RGBW_PARAMETERS", color=color)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        await self.async_action("TURN_OFF")
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        if state := self.channel_data.get("state"):
+            return state["on"]
+        return False
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        state = self.channel_data.get("state")
+        if not state or "color_brightness" not in state:
+            return None
+        brightness = state.get("color_brightness")
+        # Convert from (0-100) to Home Assistant brightness (0-255)
+        return min(255, ceil(brightness * 255 / 100.0))
```

Comment:
This seems to be 100% duplicated from the `SuplaDimmerEntity`, please move it to a shared base class.

Reply:
There's a subtle difference - `SuplaDimmerEntity` stores brightness state in `brigthness` property, while `SuplaRGBLightingEntity` stores state in `color_brightness` property. I can still extract it, and define relevant property in the subclass.